src/tests/compat_api.rs

```diff
@@ -245,247 +245,273 @@ fn start_two_services_exchange_data() {
-        &service0,
-        &service1,
-        &event_rx0,
-        &event_rx1,
-        CrustUser::Node,
-    );
-}
+        unwrap!(service0.set_accept_bootstrap(true));
 
-#[test]
-fn bootstrap_with_blacklist() {
-    let _ = env_logger::init();
+        let valid_addr = addr0.unspecified_to_localhost();
 
-    let (service0, event_rx0) = service();
+        let blacklisted_listener = unwrap!(std::net::TcpListener::bind(addr!("0.0.0.0:0")));
+        let blacklisted_addr = unwrap!(blacklisted_listener.local_addr());
+        let blacklisted_addr = PaAddr::Tcp(blacklisted_addr).unspecified_to_localhost();
+        unwrap!(blacklisted_listener.set_nonblocking(true));
 
-    unwrap!(service0.start_listening());
-    let _addr = expect_event!(event_rx0, Event::ListenerStarted(addr0) => addr0);
-    let addr0 = expect_event!(event_rx0, Event::ListenerStarted(addr0) => addr0);
+        let config1 = unwrap!(ConfigFile::new_temporary());
+        unwrap!(config1.write()).bootstrap_cache_name = Some(util::bootstrap_cache_tmp_file());
+        unwrap!(config1.write()).hard_coded_contacts = vec![
+            PeerInfo::new(valid_addr, service0.public_id()),
+            PeerInfo::with_rand_key(blacklisted_addr),
+        ];
+        let (service1, event_rx1) = service_with_config(config1);
 
-    unwrap!(service0.set_accept_bootstrap(true));
+        let uid0 = service0.public_id();
+        let uid1 = service1.public_id();
 
-    let valid_addr = addr0.unspecified_to_localhost();
+        unwrap!(service1.start_bootstrap(hashset!{blacklisted_addr}, CrustUser::Client));
 
-    let blacklisted_listener = unwrap!(std::net::TcpListener::bind(addr!("0.0.0.0:0")));
-    let blacklisted_addr = unwrap!(blacklisted_listener.local_addr());
-    let blacklisted_addr = PaAddr::Tcp(blacklisted_addr).unspecified_to_localhost();
-    unwrap!(blacklisted_listener.set_nonblocking(true));
+        expect_event!(event_rx0, Event::BootstrapAccept(id, CrustUser::Client) => {
+            assert_eq!(id, uid1);
+        });
 
-    let config1 = unwrap!(ConfigFile::new_temporary());
-    unwrap!(config1.write()).bootstrap_cache_name = Some(util::bootstrap_cache_tmp_file());
-    unwrap!(config1.write()).hard_coded_contacts = vec![
-        PeerInfo::new(valid_addr, service0.public_id()),
-        PeerInfo::with_rand_key(blacklisted_addr),
-    ];
-    let (service1, event_rx1) = service_with_config(config1);
+        expect_event!(event_rx1, Event::BootstrapConnect(id, _) => {
+            assert_eq!(id, uid0);
+        });
 
-    let uid0 = service0.public_id();
-    let uid1 = service1.public_id();
+        exchange_messages(
+            &service0,
+            &service1,
+            &event_rx0,
+            &event_rx1,
+            CrustUser::Client,
+        );
+
+        thread::sleep(Duration::from_secs(1));
+        let res = blacklisted_listener.accept();
+        match res {
+            Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => (),
```

Comment:
maybe interrupt error too ? They should probably go together most of the time. If you agree check in other places too pls

Reply:
@povilasb noted
> EINTR (`ErrorKind::Interrupted`) is retried automatically in Rust, see: https://github.com/rust-lang/rust/issues/11214